File: algorithms/cpp/sqrtx/sqrtx.cpp
// Source: https://leetcode.com/problems/sqrtx/
// Author: Miao Zhang
// Date:   2021-01-13

class Solution {
public:
    int mySqrt(int x) {
        int l = 0;
        int r = x;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (mid * mid == x) {
                return mid;
            } else if (mid * mid > x) {
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        return l - 1;
    }
};

File: algorithms/python/sqrtx/sqrtx.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/sqrtx/
# Author: Miao Zhang
# Date:   2021-01-13

class Solution:
    def mySqrt(self, x: int) -> int:
        l = 0
        r = x
        while l <= r:
            mid = (l + r) // 2
            if mid * mid == x:
                return mid
            elif mid * mid > x:
                r = mid - 1
            else:
                l = mid + 1
        return l - 1
